methods.py

```diff
@@ -81,6 +80,34 @@ def add_source_files(self, sources, files, allow_gen=False):
     return True
 
 
+def redirect_emitter(target, source, env):
+    """
+    Emitter to automatically redirect object/library build files to the `bin/obj` directory,
+    retaining subfolder structure. External build files will attempt to retain subfolder
+    structure relative to their environment's parent directory, sorted under `bin/obj/external`.
+    If `redirect_build_objects` is `False`, or an external build file isn't relative to the
+    passed environment, this emitter does nothing.
+    """
+    if not env["redirect_build_objects"]:
+        return target, source
+    redirected_targets = []
+    for item in env.Flatten(target):
```

Comment:
What exactly is the expected structure of `target`? It's surprising to me it's a deep list, and it's especially surprising you are flattening it, destroying existing nesting.

Reply:
Come to think of it, this probably doesn't do anything in this context. By this point, `target` should already be a `Node` list, so I should be able to iterate on it as-is.